src/frontend/XAPI/XAPIStatement.ts

```diff
@@ -77,8 +80,42 @@ export const verbDefinition: {
 };
 
 export class XAPIStatement {
+  static toFixed(rawNumber: number, length: number = 3): number {
+    return parseFloat(rawNumber.toFixed(length));
+  }
+
+  private startSegments: number[] = [];
+  private endSegments: number[] = [];
+  // tslint:disable-next-line:variable-name
+  private _duration: number = 0;
+
   constructor(private jwt: string, private sessionId: string) {}
 
+  set duration(duration: number) {
+    if (this._duration > 0) {
+      throw new Error('duration is already set. You can not modify it anymore');
+    }
+
+    if (duration <= 0) {
+      throw new Error('duration must be strictly positive');
+    }
+
+    this._duration = duration;
+  }
+
+  get playedSegment(): string {
```

Comment:
ditto for this please. Also needs some comments: you're making interfaces here, docstrings are probably necessary